pyroma/tests.py

```diff
@@ -248,47 +205,47 @@ def test_markdown(self):
 
 
 class PyPITest(unittest.TestCase):
-    def test_distribute(self):
-        real_urlopen = urllib.urlopen
-        real_server_proxy = xmlrpclib.ServerProxy
-        try:
-            xmlrpclib.ServerProxy = ProxyStub(
-                "distributedata.py", xmlrpclib.ServerProxy, False
-            )
-            urllib.urlopen = urlopenstub
-            data = pypidata.get_data("distribute")
-            rating = rate(data)
-
-            self.assertEqual(
-                rating,
-                (
-                    9,
-                    [
-                        "The classifiers should specify what minor versions of Python "
-                        "you support as well as what major version.",
-                        "You should have three or more owners of the project on PyPI.",
-                    ],
-                ),
-            )
-        finally:
-            xmlrpclib.ServerProxy = real_server_proxy
-            urllib.urlopen = real_urlopen
+    @unittest.mock.patch("xmlrpc.client.ServerProxy", proxystub)
+    @unittest.mock.patch("pyroma.pypidata._get_project_data")
+    def test_distribute(self, projectdatamock):
+        datafile = resource_filename(
+            __name__, os.path.join("testdata", "jsondata", "distribute.json")
+        )
+        with open(datafile, "rt") as file:
```

Comment:
```suggestion
        with open(datafile, "rt", encoding="UTF-8") as file:
```

Even if the file happens to be pure-ASCII, as mentioned previously this still needlessly breaks the tests on any system where the encoding returned by `locale.getpreferredencoding()` doesn't happen to be a strict superset of ASCII. This is the case on many Windows systems by default, e.g. for any system set to use a Japanese locale this will default to Shift-JIS, which differs from ASCII in not containing the tilde `~` and backslash `\` characters, both of which are present in these files.

Furthermore, the encoding of the input files is explicitly known (UTF-8, as required by the JSON standard and universal in practice), so it should be explicitly specified rather relying on arbitrary, environment-specific configuration—per the zen of Python, "explicit is better than implicit". And on a practical note, I find that keeping a consistent good habit of always explicitly passing the encoding of text-mode files (either as a known value or detected/inferred via some other means) takes way less effort than reasoning out every time whether it would happen to work without it, not to mention eliminates the possibility of making a mistake in that determination :)

Reply:
FYI, [PEP 597](https://www.python.org/dev/peps/pep-0597/) for Python 3.10 will make not specifying encoding a warning, due to the number of bugs this causes, for the reasons mentioned.